### PR TITLE
Relaxing Pydantic dependencies

### PR DESCRIPTION
**Pull request by lemonhead94**
This should allow for having a custom Langchain Runtime in MLServer [Issue 1920](https://github.com/SeldonIO/MLServer/issues/1920).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -62,8 +62,8 @@ gevent = "*"
 aiofiles = "*"
 orjson = "*"
 uvloop = {version = "*", markers = "sys_platform != 'win32' and (sys_platform != 'cygwin' and platform_python_implementation != 'PyPy')"}
-pydantic = "2.7.1"
-pydantic-settings = "2.2.1"
+pydantic = "^2.7.1"
```
Comment (sakoush) [inline pyproject.toml:65]: Just to note that I dont think there is anything specific to pydantic ver 2.7.1 in the codebase, I think that we support pydantic v2 in general and probably this specific version was pinned at the time of testing an upgrade. It is fine to keep it as such though I think just to be on the safe side.

**Review by sakoush — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (lemonhead94) [reply pyproject.toml:65]: Can you rerun the xgboost test, it seems to be fluke

Comment (sakoush) [reply pyproject.toml:65]: sure, they pass now.

**Comment by sakoush** [comment]
@lemonhead94 thanks for your contribution, there are related changes to move to ver 2.9 for pydantic done in https://github.com/SeldonIO/MLServer/pull/1934

**Pull request merged**